app/upgrades.go

```diff
@@ -230,6 +232,37 @@ func removeP8eMemorializeContractFee(ctx sdk.Context, app *App) {
 	}
 }
 
+// removeInactiveValidatorDelegations unbonds all delegations from inactive validators, triggering their removal from the validator set.
+func removeInactiveValidatorDelegations(ctx sdk.Context, app *App) {
+	unbondingTimeParam := app.StakingKeeper.GetParams(ctx).UnbondingTime
+	ctx.Logger().Info(fmt.Sprintf("removing all delegations from validators that have been inactive (unbonded) for %d days", int64(unbondingTimeParam.Hours()/24)))
+	removalCount := 0
+	validators := app.StakingKeeper.GetAllValidators(ctx)
+	for _, validator := range validators {
+		if validator.IsUnbonded() {
+			inActiveDuration := ctx.BlockTime().Sub(validator.UnbondingTime)
+			if inActiveDuration >= unbondingTimeParam {
+				ctx.Logger().Info(fmt.Sprintf("validator %v has been inactive (unbonded) for %d days and will be removed", validator.OperatorAddress, int64(inActiveDuration.Hours()/24)))
+				valAddress, err := sdk.ValAddressFromBech32(validator.OperatorAddress)
+				if err != nil {
+					panic(err)
+				}
+				delegations := app.StakingKeeper.GetValidatorDelegations(ctx, valAddress)
+				for _, delegation := range delegations {
+					ctx.Logger().Info(fmt.Sprintf("undelegate delegator %v from validator %v of all shares (%v)", delegation.DelegatorAddress, validator.OperatorAddress, delegation.GetShares()))
+					// unbonding all of a delegators shares, method will remove validator from set once all have been removed.
```

Comment:
this comment reads weird with the new streamlined implementation. 
```suggestion

```

Reply:
I'm open to suggestion on changing them.  I went back and forth between a lot of different messages.